### ISSUE TITLE
Workaround for possible nvim bug that makes filetype detection fail in certain conditions

### ISSUE THREAD

**Issue by abigagli** [AstroNvim/AstroNvim#1562]
### Is your feature related to a problem?

I've reported an issue in nvim-dap-ui [here](https://github.com/rcarriga/nvim-dap-ui/issues/219#issue-1560971785) because filetype detection stopped working after a recent commit of theirs.
But it turns out this seems related to some race-y behaviour possibly caused by some nvim bug, which seems to be workable-around by changing how nvim-dap-ui setup is invoked

### Describe the new feature

As reported in the linked discussion, they suggest

> To workaround the issue, you can wrap dapui.setup in a vim.schedule call:

  ```
    vim.schedule(function()
        require("dapui").setup()
      end)
```

### Additional context

I wonder if such workaround could be applied to Astronvim to alleviate the problem?

**Comment by mehalter** [comment]
@abigagli thanks for reporting! We actually do wrap all of our plugin loading in the upcoming v3 release with `vim.schedule` which is good! Thankfully this new commit doesn't affect the current stable release of AstroNvim, but this has been taken care of in v3.

**Comment by mehalter** [comment]
Since this has been  resolved in the new AstroNvim v3.0 and it's available on `nightly` I'm going to go ahead and close this as completed! Thanks again for reporting 😄
